### ISSUE TITLE
Possible problem tallying photon heating with weight windows enabled

### ISSUE THREAD

**Issue by pshriwise** [openmc-dev/openmc#3207]
## Bug Description

I was staring at the section on photon heating today and noticed that we may be subtracting too much energy from photon heating tallies when weight windows are applied.

https://github.com/openmc-dev/openmc/blob/2d988a69a10dec9e2964782d20a08927574d763a/src/tallies/tally_scoring.cpp#L958-L981

Based on how the `n_bank_second` attribute is incremented in the `Particle::create_secondary` method and that this method is applied when splitting particles in weight window application, it seems particles split in application of weight windows will be incorporated into the correction, which I don't believe is correct. 

https://github.com/openmc-dev/openmc/blob/2d988a69a10dec9e2964782d20a08927574d763a/src/weight_windows.cpp#L116-L118

I'll update this issue when I have more information.

**Comment by giovanni-mariano** [comment]
Hello Patrick! I faced this issue as well during my tests but I haven't had much time to spend on it. 
My conclusion (I think I wrote something on slack) was that it affect all the tallies which are not using the track-length estimator: basically all the tallies using the pre-collision weight (p.wgt_last()) in the score calculation instead of the current weight (p.wgt()). But I'm not sure on 100%.
Best,
Giovanni

**Comment by pshriwise** [comment]
Yeah I need to dig into this more soon. I'll try to come up with a minimal working example to start -- hopefully a problem where this effect is particularly pronounced.

**Comment by SteSeg** [comment]
I think the problem stands. A while ago I was running some fusion benchmarks with nuclear heating as tally and I remember I got weird results. Photon heating with weight windows was indeed negative in almost all positions.

Is it possible that it only happens when also photon transport is affected by ww? I think I got the negative results only when I tried to generate ww for both neutrons and photons while ww for only neutrons seem to work fine. 

Here the results without weight windows (track-length estimator):
![image](https://github.com/user-attachments/assets/e2c44b9a-5b2d-418a-b4ba-80aa0da67245)

Here the results for the same tally but with weight windows for only neutrons (track-length estimator):
![image](https://github.com/user-attachments/assets/86c99f17-58cf-4bb7-a6b4-92854eee4cdf)

Here same tally with weight windows for both neutrons and photons (track-lenght estimator):
![image](https://github.com/user-attachments/assets/87dd79a3-5109-44f4-8cfc-727b6a2fda3c)

However, I'm not sure it is an estimator issue, these are results for the same tally with ww for heating and a collision estimator (only neutron ww, collision estimator):
![image](https://github.com/user-attachments/assets/ee9b8d27-44ef-4833-944f-acf546944cad)


Lastly, in the benchmark the case withouth ww and the case with ww only for neutrons behave very well when compared with experimental results:

![image](https://github.com/user-attachments/assets/ba4dbbb5-92b3-4910-8371-1fea092ceaef)

**Comment by pshriwise** [comment]
Super helpful!! Thank you @SteSeg.
